### ISSUE TITLE
Color TV (NTSC) video mode not doing vertical color bleeding

### ISSUE THREAD

**Issue by tomcw** [AppleWin/AppleWin#344]
This is a regression from AppleWin 1.25.0.3.
Demonstrable using Fantavision, Blazing Paddles, Karateka, Archon title (grey on cube & cage), Aztec title, etc.

**Comment by tomcw** [comment]
Archon title 1.25.0.3 (TV mode)
![applewin_screenshot_000000001](https://cloud.githubusercontent.com/assets/6696896/18600629/9d49f374-7c56-11e6-9602-1f8cfd2029cc.png)

Archon title 1.26.01 (Color TV mode)
![archon_i_000000000](https://cloud.githubusercontent.com/assets/6696896/18600697/f57daefa-7c56-11e6-8b86-d3ec23bfe3a8.png)

**Comment by Michaelangel007** [comment]
Isn't `TV` mode a hack anyways?

Does anyone have a picture of a real CRT TV so we can compare against these two?

I'm concerned that rendering would take another performance hit as "mixing" to grey is not trivial WRT Apple's colors since we no longer have 16 colors  but 64 x 256 = 16,384 !!!

I guess we should which colors are being mixed ... there might be a way to optimize the mixing ...

**Comment by Michaelangel007** [comment]
Some thoughts on optimizing the mixing ..

* Take the high nibbles and use that as an index
* Since the apple can only display 4,096 colors extract THOSE two indices and use that as the index
* Run the mixer on a different thread

**Comment by tomcw** [comment]
> Isn't  TV  mode a hack anyways?

No, IMO, since games & paint packages took advantage of it (ie. the vertical bleeding), as another example, look at AZTEC.DSK's title screen: eg. the pink wall, violet hat rim.

MJ Mahon [suggests](https://groups.google.com/forum/?hl=en#!topic/comp.emulators.apple2/lCsi6oziNAo) that it a function of the human eye, but perhaps purely from the POV of monitor not TV emulation.

**Comment by tomcw** [comment]
1.25.0.4, "TV":

![applewin_screenshot_000000000](https://cloud.githubusercontent.com/assets/6696896/18681108/807cda08-7f5e-11e6-945d-28ccad50de6b.png)

1.26.0.3, "Color TV":

![applewin_screenshot_000000001](https://cloud.githubusercontent.com/assets/6696896/18681121/89d3b086-7f5e-11e6-8dc3-9857e7cd1267.png)

**Comment by Michaelangel007** [comment]
Here's MJ Mahon's quote (emphasis mine):

```
Blending of adjacent scan lines is more a function of the human eye than 
the monitor.  I **don't** see it as an appropriate emulator function. 
```

**Comment by Michaelangel007** [comment]
I _see_ what you mean Tom. (See what I did there? Ah crap, these puns will never stop. :-) )

Anyways, with those "thumbnails" the _intent_ is pretty clear.

* Hat Rim = Purple
* Wall = pink

For comparison here is 1.26.03 with:

* Even scanlines removed
![aztec_even](https://cloud.githubusercontent.com/assets/7876796/18682589/b8be1950-7f21-11e6-99a3-7ee44647f163.png)

* Odd scanlines removed
![aztec_odd](https://cloud.githubusercontent.com/assets/7876796/18682594/be16f4b2-7f21-11e6-8086-687d4969a142.png)

Crystal clear proof.  

Make me wonder if the artists were are of Edwin H. Land's (Polaroid) Retinex Theory ?!
See [pages](http://www.slideserve.com/holly/retinex-theory) 4 & 5 to have your mind blown.

**Comment by tomcw** [comment]
From cea2, "AppleWin 1.26.0 pre-release" thread:

**Comment by tomcw** [comment]
For reference, here's The Bard's Tale intro screen, which (for the 2 hats) mixes:
- Orange+Purple=Pink (like Aztec above)
- Green+Blue=Cyan

1.25.0.4: 50% TV Mode
![the bard s tale_1 25 0 4-50](https://user-images.githubusercontent.com/6696896/44948536-c1477780-ae16-11e8-9de6-036b35d23715.png)

1.25.0.4: 100% TV Mode
![the bard s tale_1 25 0 4-100](https://user-images.githubusercontent.com/6696896/44948538-c4426800-ae16-11e8-90d2-d35199b4cad2.png)

1.27.7: 50% Color TV Mode
![the bard s tale_1 27 7-50](https://user-images.githubusercontent.com/6696896/44948539-cd333980-ae16-11e8-9dfa-22a50dae8545.png)

1.27.7: 100% Color TV Mode
![the bard s tale_1 27 7-100](https://user-images.githubusercontent.com/6696896/44948540-d15f5700-ae16-11e8-94e6-55f43d781062.png)

Real Apple //e, PAL, outputting to a Mitsubishi CT-1535TX CRT TV (~1989?) via a Digitek TV modulator.
My poor quality(!) screen-grab from a video using a Sony HDR-PJ620 HandyCam.
![bt-video2-avchd-38sec-s](https://user-images.githubusercontent.com/6696896/44948550-24390e80-ae17-11e8-97ca-ae1ada694024.png)

NB. The NTSC rendering for 1.27.7 should be the same as 1.26 (ie. AFAICR, I've not changed anything).
Also AppleWin only supports NTSC and not PAL, so the above emulated NTSC vs real PAL comparison isn't particularly fair.

**Comment by tomcw** [comment]
More pictures, but this time taken using a Samsung (WB250F) point-and-click digit camera:

![sam_2863-s](https://user-images.githubusercontent.com/6696896/44955964-11284c00-aeb4-11e8-9b11-a5202552e092.png)

![sam_2857-s](https://user-images.githubusercontent.com/6696896/44955973-3917af80-aeb4-11e8-876a-750f0233c7b7.png)

![sam_2858-s](https://user-images.githubusercontent.com/6696896/44955974-3ae17300-aeb4-11e8-96be-64aa14895e2e.png)

**Comment by tomcw** [comment]
Some more pictures (same as before, ie: //e, PAL; Samsung digital camera)

Archon:
NB. AppleWin 1.25.0.4 mixes purple+green to grey!

![sam_2866-s](https://user-images.githubusercontent.com/6696896/44956001-af1c1680-aeb4-11e8-877d-b7da800f581a.png)

Fantavision's palette-1:

![sam_2849-s](https://user-images.githubusercontent.com/6696896/44956013-e38fd280-aeb4-11e8-9da2-6f98392eacc5.png)

Karateka (blue+orange=green ground):
NB. AppleWin 1.25.0.4 mixes blue+orange to grey!

![sam_2868-s](https://user-images.githubusercontent.com/6696896/44956032-3ec1c500-aeb5-11e8-8b14-d41674eca56d.png)
